encrypter.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+#include <utility>
+
+using Byte = char;
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+void shuffle(std::vector<Byte>& input)
+{
+  std::default_random_engine gen;
+
+  int i = 0;
+  for (std::vector<Byte>::iterator iter = input.begin(); iter < input.end(); ++i, ++iter)
+  {
+    std::uniform_int_distribution<int> dis(i, input.size() - 1);
+    std::swap(input[i], input[dis(gen)]);
+  } 
+}
+
+//=====replace_extension=====//
+std::string replace_extension(const std::string& filename, const std::string& new_extension)
+{
+  int i = filename.rfind('.');
```

Comment:
variable i is not so local to have such short name. It should be more descriptive.